backend/src/main/java/com/woowacourse/momo/group/service/GroupSearchService.java

```diff
@@ -35,26 +38,37 @@ public class GroupSearchService {
     private final GroupFindService groupFindService;
     private final GroupSearchRepository groupSearchRepository;
     private final FavoriteRepository favoriteRepository;
+    private final GroupImageRepository groupImageRepository;
+    private final ImageProvider imageProvider;
 
     public GroupResponse findGroup(Long groupId) {
         Group group = groupFindService.findByIdWithHostAndSchedule(groupId);
-        return GroupResponseAssembler.groupResponse(group);
+        String imageUrl = getImageUrl(group);
+        return GroupResponseAssembler.groupResponse(group, imageUrl);
+    }
+
+    private String getImageUrl(Group group) {
+        GroupImage groupImage = groupImageRepository.findByGroupId(group.getId())
+                .orElseGet(() -> new GroupImage(group.getId(), group.getCategory().getDefaultImageName()));
+        String imageName = groupImage.getImageName();
```

Comment:
```suggestion
        String imageName = groupImageRepository.findByGroupId(group.getId())
                .map(GroupImage::getImageName)
                .orElse(group.getCategory().getDefaultImageName());
```
다음과 같이 리팩토링하면 깔끔할 것 같아요!!